internal/ssm/registration.go

```diff
@@ -0,0 +1,143 @@
+package ssm
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	awsSsm "github.com/aws/aws-sdk-go-v2/service/ssm"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+)
+
+const registrationFilePath = "/var/lib/amazon/ssm/registration"
+
+type SSMRegistration struct {
+	installRoot string
```

Comment:
If we are gonna introduce an interface for registration and mock it in the tests, do we even need this? I see that you use this from the mock itself, but that's kind of weird. We are testing the implementation of its methods by reusing it to implement a mock. vs letting the code using it directly.

I still like this field and testing the real deletion though. Maybe what we can do is avoid mocking the whole registration? It seems like the only reason we need to mock it is because it returns a client. So what if we just inject the client as another dependency to de-register? we can then only mock the client and use `SSMRegistration` directly